### PR TITLE
fix: ADD_IF_ABSENT Action Saves Files to User Bucket Instead of Publi…

### PR DESCRIPTION
**Pull request by astsiapanay**
…c Folder #569

### CODE REVIEW

File: server/src/main/java/com/epam/aidial/core/server/service/PublicationService.java [hunk 1]
```diff
@@ -223,6 +224,10 @@ public Publication approvePublication(ResourceDescriptor resource) {
                 .filter(i -> i.getAction() == Publication.ResourceAction.ADD)
                 .toList();
 
+        List<Publication.Resource> resourcesToAddIfAbsent = publication.getResources().stream()
```
Comment (artsiomkorzun) [inline server/src/main/java/com/epam/aidial/core/server/service/PublicationService.java]: Please check with checkReviewResources(resourcesToAddIfAbsent).

And delete deleteReviewResources(resourcesToAddIfAbsent).

File: server/src/main/java/com/epam/aidial/core/server/service/PublicationService.java [hunk 2]
```diff
@@ -251,7 +256,10 @@ public Publication approvePublication(ResourceDescriptor resource) {
 
         ruleService.storeRules(publication);
 
-        copyReviewToTargetResources(resourcesToAdd);
+        List<Publication.Resource> allResourcesToAdd = new ArrayList<>();
+        allResourcesToAdd.addAll(resourcesToAddIfAbsent);
+        allResourcesToAdd.addAll(resourcesToAdd);
+        copyReviewToTargetResources(allResourcesToAdd);
         deleteReviewResources(resourcesToAdd);
```
Comment (artsiomkorzun) [inline server/src/main/java/com/epam/aidial/core/server/service/PublicationService.java]: Should you take all ADD and ADD_IF_ABSENT everywhere to make it consistent? Right now some methods take, some not.

File: server/src/main/java/com/epam/aidial/core/server/service/PublicationService.java (hunk 1 above)
Comment (astsiapanay) [reply server/src/main/java/com/epam/aidial/core/server/service/PublicationService.java]: fixed

File: server/src/main/java/com/epam/aidial/core/server/service/PublicationService.java (hunk 2 above)
Comment (astsiapanay) [reply server/src/main/java/com/epam/aidial/core/server/service/PublicationService.java]: fixed

**Review by artsiomkorzun — APPROVED** [review]

**Comment by astsiapanay** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-core-pr-570.nightly-test.deltixhub.io
>E2E tests status: success

**Pull request merged**